File: src/modules/terminal/block/lib/history.ts
import { invoke } from "@tauri-apps/api/core";

export async function historySuggest(line: string): Promise<string | null> {
  try {
    return await invoke<string | null>("history_suggest", { line });
  } catch {
    return null;
  }
}

export async function historyCommands(
  prefix: string,
  limit = 50,
): Promise<string[]> {
  try {
    return await invoke<string[]>("history_commands", { prefix, limit });
  } catch {
    return [];
  }
}

export async function historyList(
  query: string,
  limit = 100,
): Promise<string[]> {
  try {
    return await invoke<string[]>("history_list", { query, limit });
  } catch {
    return [];
  }
}

export async function historyRecord(command: string): Promise<void> {
  try {
    await invoke<void>("history_record", { command });
  } catch {
    // silent
  }
}

File: src-tauri/src/modules/pty/history.rs
/// Shell history command stubs.
/// These return empty results until a full SQLite-backed history implementation
/// is added in a future iteration.

#[tauri::command]
pub async fn history_suggest(_line: String) -> Result<Option<String>, String> {
    Ok(None)
}

#[tauri::command]
pub async fn history_commands(_prefix: String, _limit: usize) -> Result<Vec<String>, String> {
    Ok(vec![])
}

#[tauri::command]
pub async fn history_list(_query: String, _limit: usize) -> Result<Vec<String>, String> {
    Ok(vec![])
}

#[tauri::command]
pub async fn history_record(_command: String) -> Result<(), String> {
    Ok(())
}
